plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+      return None
+
+    general_keystore_file_object = general_keystore_entry.GetFileObject()
+    general_keystore_data = general_keystore_file_object.read()
+    json_data = json.loads(general_keystore_data)
+
+    if not json_data:
+      parser_mediator.ProduceExtractionWarning(
+          'Unable to parse OneDrive keystore file {0:s}'.format(
+              general_keystore_entry.name))
+      return None
+
+    if not json_data or not isinstance(json_data, list):
+      return None
+
+    key = json_data[0].get('Key', None)
+    if not key:
+      return None
+
+    decoded_key = base64.b64decode(key)
+
+    if len(decoded_key) != 32:
+      return None
+
+    return decoded_key
+
+  def _DeobfuscateStrings(self, extracted_strings, obfuscated_string_map):
+    """Deobfuscate strings extracted from OneDrive logs.
+
+    Args:
+      extracted_strings (list[str]): strings extracted from OneDrive log files.
+      obfuscated_string_map (dict[str, str]): the parsed obfuscated string map.
+
+    Returns:
+      list[str]: the list of deobfuscated strings.
+    """
+    if not obfuscated_string_map:
+      return extracted_strings
+
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+      output_string = []
+      for token in tokens:
+        if token not in self._DELIMITERS:
+          token = obfuscated_string_map.get(token, token)
+        output_string.append(token)
+      output_strings.append(''.join(output_string))
+
+    return output_strings
+
+  def _DecryptStrings(self, extracted_strings, aes_key):
+    """Decrypt strings extracted from OneDrive logs.
+
+    Encrypted strings are serialized as follows:
+    * padded using PKCS7 then encrypted using AES in CBC mode.  The AES key is
+      from the general.keystore file and the initialization vector is null.
+    * URL safe base64 encoded.
+
+    Args:
+      extracted_strings (list[str]): strings extracted from OneDrive log files.
+      aes_key (bytes): the AES key.
+
+    Returns:
+      list[str]: the decrypted strings.
+    """
+    if not aes_key:
+      return extracted_strings
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+
+      output_string = []
+      for token in tokens:
+        if token in self._DELIMITERS:
+          output_string.append(token)
+          continue
+
+        token_length = len(token)
+
+        # minimum block size of base64 encoded AES-CBC is 22 bytes
+        if token_length < 22:
+          output_string.append(token)
+          continue
+
+        if not all(character in self._BASE64_ALPHABET for character in token):
+          output_string.append(token)
+          continue
+
+        rem = token_length % 4
+        if rem == 1:
+          output_string.append(token)
+          continue
+
+        while len(token) % 4:
+          token += '='
```

Comment:
please don't use `+` for string concatenation (not efficient) why not use 

```
base64_padding_size = len(token) % 4
token = ''.join([token, '=' * base64_padding_size])

Reply:
Done